src/commands.rs

```diff
@@ -4239,17 +4240,44 @@ fn cmd_git_push(
                 ));
             }
             Some(checkout) => {
-                let desired_target = Some(RefTarget::Normal(checkout.clone()));
-                for (branch_name, branch_target) in workspace_command.repo().view().branches() {
-                    if branch_target.local_target == desired_target {
-                        let push_action = classify_branch_push_action(branch_target, &args.remote);
-                        match push_action {
-                            BranchPushAction::AlreadyMatches => {}
-                            BranchPushAction::LocalConflicted => {}
-                            BranchPushAction::RemoteConflicted => {}
-                            BranchPushAction::Update(update) => {
-                                branch_updates.push((branch_name.clone(), update));
-                            }
+                fn find_branches_targeting<'a>(
+                    view: &'a View,
+                    target: &RefTarget,
+                ) -> Vec<(&'a String, &'a BranchTarget)> {
+                    view.branches()
+                        .iter()
+                        .filter(|(_, branch_target)| {
+                            branch_target
+                                .local_target
+                                .as_ref()
+                                .map_or(false, |x| x == target)
```

Comment:
`branch_target.local_target.as_ref() == Some(target)` is a little simpler. I don't know which is faster, but speed probably doesn't matter here anyway. What do you think?

Reply:
Yep, that's definitely better.